.ci/compute_projects.py

```diff
@@ -0,0 +1,186 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Computes the list of projects that need to be tested from a diff.
+
+Does some things, spits out a list of projects.
+"""
+
+from collections.abc import Set
+import pathlib
+import platform
+import sys
+
+PROJECT_DEPENDENCIES = {
+    "llvm": set(),
+    "clang": {"llvm"},
+    "bolt": {"clang", "lld", "llvm"},
+    "clang-tools-extra": {"clang", "llvm"},
+    "compiler-rt": {"clang", "lld"},
+    "libc": {"clang", "lld"},
+    "openmp": {"clang", "lld"},
+    "flang": {"llvm", "clang"},
+    "lldb": {"llvm", "clang"},
+    "libclc": {"llvm", "clang"},
+    "lld": {"llvm"},
+    "mlir": {"llvm"},
+    "polly": {"llvm"},
+}
+
+DEPENDENTS_TO_TEST = {
+    "llvm": {"bolt", "clang", "clang-tools-extra", "lld", "lldb", "mlir", "polly"},
+    "lld": {"bolt", "cross-project-tests"},
+    "clang": {"clang-tools-extra", "compiler-rt", "cross-project-tests"},
+    "clang-tools-extra": {"libc"},
+    "mlir": {"flang"},
+}
+
+DEPENDENT_RUNTIMES_TO_TEST = {"clang": {"libcxx", "libcxxabi", "libunwind"}}
```

Comment:
[PEP 8](https://peps.python.org/pep-0008/#global-variable-names), which we [follow](https://llvm.org/docs/CodingStandards.html#python-version-and-source-code-formatting), says that global variable names should be named in the same way as functions, which are `snake_case` in this file.